src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const getUser = (id: number): User | null => {
+  return usersFromServer.find(user => user.id === id) || null;
+};
+
+const todoList: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: getUser(todo.userId),
+  }
+));
+
+const getTodoId = (todos: Todo[]): number => {
+  return Math.max(...todos.map(todo => todo.id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todoList);
+  const [userError, showUserError] = useState(false);
```

Comment:
Consider using a more descriptive name for `showUserError` and `showTitleError`. For example, `setUserErrorVisible` and `setTitleErrorVisible`. 🤔